File: backend/src/utils/response.ts
import { Response } from 'express';
import { ApiResponse } from '@shared/types';

export const success = <T>(res: Response, data?: T, message = 'success', code = 200) => {
  const result: ApiResponse<T> = {
    code,
    message,
    data,
  };
  return res.status(code).json(result);
};

export const error = (res: Response, message: string, code = 400) => {
  const result: ApiResponse = {
    code,
    message,
  };
  return res.status(code).json(result);
};

export const serverError = (res: Response, message = 'Internal Server Error', code = 500) => {
  const result: ApiResponse = {
    code,
    message,
  };
  return res.status(code).json(result);
};

File: backend/src/utils/response.py
from typing import Any, Optional
from pydantic import BaseModel


class ApiResponse(BaseModel):
    success: bool
    data: Optional[Any] = None
    message: Optional[str] = None


def success_response(data: Any = None, message: str = "操作成功") -> ApiResponse:
    return ApiResponse(success=True, data=data, message=message)


def error_response(message: str = "操作失败", data: Any = None) -> ApiResponse:
    return ApiResponse(success=False, data=data, message=message)
